src/Runner.Worker/Handlers/ScriptHandler.cs

```diff
@@ -222,10 +223,19 @@ public async Task RunAsync(ActionRunStage stage)
             {
                 throw new ArgumentException("Invalid shell option. Shell must be a valid built-in (bash, sh, cmd, powershell, pwsh) or a format string containing '{0}'");
             }
-
-            // We do not not the full path until we know what shell is being used, so that we can determine the file extension
-            var scriptFilePath = Path.Combine(tempDirectory, $"{Guid.NewGuid()}{ScriptHandlerHelpers.GetScriptFileExtension(shellCommand)}");
-            var resolvedScriptPath = $"{StepHost.ResolvePathForStepHost(scriptFilePath).Replace("\"", "\\\"")}";
+            string scriptFilePath, resolvedScriptPath;
+            bool shouldWriteContentToDisk = !Inputs.ContainsKey("path");
+            if (shouldWriteContentToDisk)
```

Comment:
should we also branch based on `IsActionStep`?

Reply:
I thought about it, but wanted to keep this generic in the sense that if we eventually add the ability to run a file from your repo rather then a `run:` step as a first class feature we could just reuse this.

We don't have those scenarios today, but if this evolves over time it would be nice to just have this function support it as is.

Reply:
We don't, but I will hide this behind that flag anyways